store/sqlstore/status_store.go

```diff
@@ -55,26 +55,24 @@ func (s SqlStatusStore) SaveOrUpdate(status *model.Status) store.StoreChannel {
 	})
 }
 
-func (s SqlStatusStore) Get(userId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		var status model.Status
+func (s SqlStatusStore) Get(userId string) (*model.Status, *model.AppError) {
+	var status model.Status
 
-		if err := s.GetReplica().SelectOne(&status,
-			`SELECT
+	if err := s.GetReplica().SelectOne(&status,
+		`SELECT
 				*
 			FROM
 				Status
 			WHERE
 				UserId = :UserId`, map[string]interface{}{"UserId": userId}); err != nil {
-			if err == sql.ErrNoRows {
-				result.Err = model.NewAppError("SqlStatusStore.Get", MISSING_STATUS_ERROR, nil, err.Error(), http.StatusNotFound)
-			} else {
-				result.Err = model.NewAppError("SqlStatusStore.Get", "store.sql_status.get.app_error", nil, err.Error(), http.StatusInternalServerError)
-			}
+		if err == sql.ErrNoRows {
+			return nil, model.NewAppError("SqlStatusStore.Get", MISSING_STATUS_ERROR, nil, err.Error(), http.StatusNotFound)
 		} else {
```

Comment:
You don't need the else, because the if always `return`, just put the content one level up and remove the else.